File: src/mysqlcookbook/recipes/api/05_null_in_result/NullInResult.java
import java.sql.*;
import com.kitebird.mcb.Cookbook;

public class NullInResult
{
  public static void main (String[] args)
  {
    Connection conn = null;

    try
    {
      conn = Cookbook.connect ();
      System.out.println ("Connected");

      // for queries that do return a result set
      // (permits NULL values in data)
      {
      try
      {
//#@ _FETCHLOOP_
        Statement s = conn.createStatement ();
        s.executeQuery ("SELECT name, birth, foods FROM profile");
        ResultSet rs = s.getResultSet ();
        ResultSetMetaData md = rs.getMetaData ();
        int ncols = md.getColumnCount ();
        while (rs.next ())  // loop through rows of result set
        {
          for (int i = 0; i < ncols; i++) // loop through columns
          {
            String val = rs.getString (i+1);
            if (i > 0)
              System.out.print (", ");
            if (rs.wasNull ())
              System.out.print ("NULL");
            else
              System.out.print (val);
          }
          System.out.println ();
        }
        rs.close ();  // close result set
        s.close ();   // close statement
//#@ _FETCHLOOP_
      }
      catch (Exception e)
      {
        Cookbook.printErrorMessage (e);
      }
      }

    }
    catch (Exception e)
    {
      System.err.println ("Cannot connect to server");
    }
    finally
    {
      if (conn != null)
      {
        try
        {
          conn.close ();
          System.out.println ("Disconnected");
        }
        catch (Exception e) { }
      }
    }
  }
}

File: src/mysqlcookbook/recipes/api/05_null_in_result/null-in-result.py
#!/usr/bin/python3

import mysql.connector
import cookbook

conn = cookbook.connect()

try:
#@ _FETCHLOOP_
  cursor = conn.cursor()
  cursor.execute("SELECT name, birth, foods FROM profile")
  for row in cursor:
    row = list(row)  # convert nonmutable tuple to mutable list
    for i, value in enumerate(row):
      if value is None:  # is the column value NULL?
        row[i] = "NULL"
    print("name: %s, birth: %s, foods: %s" % (row[0], row[1], row[2]))
  cursor.close()
#@ _FETCHLOOP_
except mysql.connector.Error as e:
  print("Oops, the statement failed")
  print("Error: %s" % e)

conn.close()
